File: 新multi agent demo 5.8/backend/app/audit.py
from __future__ import annotations

from typing import Any
from uuid import uuid4

from pydantic import BaseModel, Field


class AuditRequest(BaseModel):
    content: str
    audit_id: str | None = None
    config: dict[str, Any] | None = None


class BlackboardEntry(BaseModel):
    timestamp: str
    agent: str
    zone: str
    phase: str
    content: str
    data: dict[str, Any] = Field(default_factory=dict)
    raw_llm_output: str | None = None


class AuditResult(BaseModel):
    audit_id: str = Field(default_factory=lambda: str(uuid4()))
    content: str
    final_verdict: str  # violation | normal | review | reject | pass
    confidence: float
    logs: list[BlackboardEntry] = Field(default_factory=list)
    agent_results: dict[str, Any] = Field(default_factory=dict)

File: 新multi agent demo 5.8/frontend/src/audit.ts
export interface BlackboardEntry {
  timestamp: string;
  agent: string;
  zone: string;
  phase: string;
  content: string;
  data?: {
    verdict?: string;
    confidence?: number;
    reason?: string;
    [key: string]: unknown;
  };
  raw_llm_output?: string;
}

export interface AuditResult {
  audit_id: string;
  content: string;
  final_verdict: "violation" | "normal" | "review" | "reject" | "pass";
  confidence: number;
  logs: BlackboardEntry[];
  agent_results: Record<string, unknown>;
}

export interface AgentConfig {
  enabled: boolean;
  zone?: string;
  temperature?: number;
  [key: string]: unknown;
}

export type ModelProvider = "openai" | "deepseek" | "moonshot" | "zhipu" | "qwen" | "custom";

export interface ModelConfig {
  id: string;
  name: string;
  provider: ModelProvider;
  api_key: string;
  base_url: string;
  small_model: string;
  strong_model: string;
  embedding_model: string;
  is_default: boolean;
  enabled: boolean;
}

export interface TestConnectionResult {
  success: boolean;
  message: string;
  models_available: string[];
}
